pkg/controllers/provisioning/v1alpha1/reallocation/terminate.go

```diff
@@ -30,39 +30,50 @@ type Terminator struct {
 	cloudprovider cloudprovider.Factory
 }
 
-func (t *Terminator) CordonNodes(ctx context.Context, nodes []*v1.Node) error {
+// AddTTLs takes in a list of underutilized nodes and adds TTL to them
+func (t *Terminator) AddTTLs(ctx context.Context, nodes []*v1.Node) error {
+	if len(nodes) == 0 {
+		return nil
+	}
 	for _, node := range nodes {
-		if node.Spec.Unschedulable {
-			continue
-		}
-		persisted := node.DeepCopyObject()
-		node.Spec.Unschedulable = true
+		persisted := node.DeepCopy()
+		node.Annotations[v1alpha1.ProvisionerTTLKey] = time.Now().Add(300 * time.Second).Format(time.RFC3339)
 		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-			return fmt.Errorf("patching node %s, %w", node.Name, err)
+			return fmt.Errorf("patching node, %w", err)
 		}
 	}
+	zap.S().Debugf("Added TTL to %d nodes", len(nodes))
```

Comment:
Is it too spammy to add a debug statement for each TTL we add?

Reply:
Do you mean for each node? I don't think would spam that much, for the useful amount of information it gives.